src/components/ItemList.js

```diff
@@ -1,41 +1,67 @@
-import { collection, query, where, onSnapshot } from 'firebase/firestore';
-import { useEffect, useState } from 'react';
-import { db } from '../lib/firebase.js';
+import { useState } from 'react';
+import useFirebaseSnapshot from '../hooks/useFirebaseSnapshot.js';
+import cleanData from '../utils/cleanData.js';
 import { Link } from 'react-router-dom';
 
 const ItemList = () => {
-  const [docs, setDocs] = useState([]);
-
-  useEffect(() => {
-    const token = localStorage.getItem('token');
-    const q = query(
-      collection(db, 'shopping-list'),
-      where('token', '==', token),
-    );
-    const unsubscribe = onSnapshot(q, (querySnapshot) => {
-      const items = [];
-      querySnapshot.forEach((doc) => {
-        items.push(doc.data());
-      });
-      setDocs(items);
+  const [searchInput, setSearchInput] = useState('');
+  const [filteredResults, setFilteredResults] = useState('');
+
+  const { docs, loading } = useFirebaseSnapshot();
+
+  const filterItems = (query) => {
+    setSearchInput(query);
+    const results = docs.filter((item) => {
+      const name = cleanData(item.name);
+      return name.includes(cleanData(query));
     });
-    return () => {
-      unsubscribe();
-    };
-  }, []);
+    setFilteredResults(results);
+  };
+
+  const handleClear = () => {
+    setSearchInput('');
+    setFilteredResults('');
+  };
 
   return (
     <>
       <h2>Smart Shopping List</h2>
-      <ul>
-        {docs.length > 0 ? (
-          docs.map((item) => <li key={item.token + item.name}>{item.name}</li>)
-        ) : (
-          <p>
-            No items yet! <Link to="/add-item">Add one.</Link>
-          </p>
-        )}
-      </ul>
+      {loading && <p>Loading ...</p>}
+
+      {!docs.length && !loading && (
+        <p>
+          No items yet! <Link to="/add-item">Add some.</Link>
+        </p>
+      )}
+
+      {docs.length > 0 && (
+        <>
+          <form>
+            <label htmlFor="filter-items">Filter Items</label>
+            <input
+              id="filter-items"
+              type="text"
+              name="filter-items"
+              value={searchInput}
+              onChange={({ target }) => filterItems(target.value)}
+            />
+            <label htmlFor="btn">clear</label>
+            <input
+              type="btn"
```

Comment:
I don't believe `btn` is a [valid](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#input_types) type for an input element. You likely mean `button` here? It's a bit of a nitpick since the input element is being hidden with CSS.

Reply:
Good catch, have resolved the issue by changing tag back to button and giving button type='button'.